queue/src/fiskaltrust.Middleware.Localization.QueueIT/SignProcessorIT.cs

```diff
@@ -2,56 +2,217 @@
+                    Moment = DateTime.UtcNow,
+                    Message = $"QueueId {queueItem.ftQueueId} has not been activated yet."
+                }
+            };
+            receiptResponse.ftState += ftStatesFlags.SECURITY_MECHAMISN_DEACTIVATED;
+            receiptResponse.ftReceiptIdentification = $"ft{queue.ftReceiptNumerator:X}#";
+            return (receiptResponse, actionJournals);
+        }
+
+        public (ReceiptResponse receiptResponse, List<ftActionJournal> actionJournals) ReturnWithQueueIsDisabled(ftQueue queue, ReceiptResponse receiptResponse, ftQueueItem queueItem)
+        {
+            var actionJournals = new List<ftActionJournal>
+            {
+                new ftActionJournal
+                {
+                    ftActionJournalId = Guid.NewGuid(),
+                    ftQueueId = queueItem.ftQueueId,
+                    ftQueueItemId = queueItem.ftQueueItemId,
+                    Moment = DateTime.UtcNow,
+                    Message = $"QueueId {queueItem.ftQueueId} has been disabled."
+                }
+            };
+            receiptResponse.ftState += ftStatesFlags.SECURITY_MECHAMISN_DEACTIVATED;
+            receiptResponse.ftReceiptIdentification = $"ft{queue.ftReceiptNumerator:X}#";
+            return (receiptResponse, actionJournals);
+        }
+
+        public async Task<(ReceiptResponse receiptResponse, List<ftActionJournal> actionJournals)> ProcessFailedReceiptRequest(ftQueueIT queueIt, ftQueueItem queueItem, ReceiptResponse receiptResponse)
+        {
+            if (queueIt.SSCDFailCount == 0)
             {
-                var requestCommandResponse = await requestCommand.ProcessFailedReceiptRequest(_signingDevice, _logger, _countrySpecificSettings, queue, queueItem, request).ConfigureAwait(false);
-                return (requestCommandResponse.ReceiptResponse, requestCommandResponse.ActionJournals.ToList());
+                queueIt.SSCDFailMoment = DateTime.UtcNow;
+                queueIt.SSCDFailQueueItemId = queueItem.ftQueueItemId;
             }
-            var response = await requestCommand.ExecuteAsync(queue, request, queueItem).ConfigureAwait(false);
-            return (response.ReceiptResponse, response.ActionJournals.ToList());
+            queueIt.SSCDFailCount++;
+            await _configurationRepository.InsertOrUpdateQueueITAsync(queueIt).ConfigureAwait(false);
+            var log = $"Queue is in failed mode. SSCDFailMoment: {queueIt.SSCDFailMoment}, SSCDFailCount: {queueIt.SSCDFailCount}.";
+            receiptResponse.ftState |= 0x2;
+            // TODO => we should probably use error state here for all receipts EEEE_EEEE, since it is not allowed to continuing operation while being in wrong mode
+            _logger.LogInformation(log);
+            receiptResponse.SetFtStateData(new StateDetail() { FailedReceiptCount = queueIt.SSCDFailCount, FailMoment = queueIt.SSCDFailMoment });
+            return (receiptResponse, new List<ftActionJournal>());
```

Comment:
In the case of the failure scenario where the printer is not reachable this is in my opinion in italy the exact same scenario as if the whole middleware is not reachable. At least from the PosCreators perspective.

In both cases they have to create handwritten receipts and send them again to us with the `0x0008` process as handwritten flag.

It makes no sense for them to also send these receipts to the middleware because they would have to write them per hand _and_ on the possystem.

When we get these receipts with the handwritten flag we just need to send them to the printer like we would normally. I think if they are older than 12 days at the time of resending the printer will throw an error (not 100% sure about that though).

So if the printer is not reachable we will have one queueitem of the first not reachable (which should return state `EEEE_EEEE`).
If they send another receipt in failed mode I think we should in the best case immediately fail with out creating queueitem and return state `FFFF_FFFF`.
If they send a zero receipt we try to end the failed mode and don't need to do anything else.

Then they can resend all of their receipts.

This means we will have one queueitem for the first receipt that failed and the "same" queueitem again later on with he handwritten flag.

Reply:
So right now we think that in the case of the printer being offline we are not allowed to procceed working, so there is nothing like an error mode in Italy it the moment :/. 

Something that we will still include is the way to provide handwritten receipts / late signing mode, but I am not yet sure about how we can handle that.